File: src/main/java/com/sfh/pokeRogueBot/phase/Phase.java
package com.sfh.pokeRogueBot.phase;

import com.sfh.pokeRogueBot.model.enums.UiMode;
import com.sfh.pokeRogueBot.model.exception.NotSupportedException;
import com.sfh.pokeRogueBot.phase.actions.PhaseAction;

public interface Phase {

    int defaultWaitAfterStage2x = 2000;

    int getWaitAfterStage2x();

    String getPhaseName();

    PhaseAction[] getActionsForGameMode(UiMode gameMode) throws NotSupportedException;
}

File: src/main/kotlin/com/sfh/pokeRogueBot/phase/Phase.kt
package com.sfh.pokeRogueBot.phase

import com.sfh.pokeRogueBot.model.enums.UiMode
import com.sfh.pokeRogueBot.model.exception.UnsupportedUiModeException

interface Phase {
    val phaseName: String

    @Throws(UnsupportedUiModeException::class)
    fun handleUiMode(uiMode: UiMode)
}

interface UiPhase : Phase

interface NoUiPhase : Phase

interface CustomPhase : Phase